arviz/stats/stats.py

```diff
@@ -1330,60 +1358,67 @@ def summary(
+                _make_ufunc(_multichain_statistics, n_output=5, ravel=False),
+                dataset,
+                input_core_dims=(("chain", "draw"),),
+                output_core_dims=tuple([] for _ in range(5)),
+            )
+            diagnostics_names: Tuple[str, ...] = (
+                "mcse_mean",
+                "mcse_sd",
+                "ess_bulk",
+                "ess_tail",
+                "r_hat",
+            )
 
-    # Combine metrics
-    metrics: List[xr.Dataset] = []
-    metric_names: List[str] = []
-    if extend:
-        metrics_: Tuple[xr.Dataset, ...]
-        metrics_names_: Tuple[str, ...] = (
-            "mean",
-            "sd",
-            f"hdi_{100 * alpha / 2:g}%",
-            f"hdi_{100 * (1 - alpha / 2):g}%",
-            "mcse_mean",
-            "mcse_sd",
-            "ess_bulk",
-            "ess_tail",
-            "r_hat",
-        )
-        if kind == "all":
-            metrics_ = (
-                mean,
-                sd,
-                hdi_lower,
-                hdi_higher,
-                mcse_mean,
-                mcse_sd,
-                ess_bulk,
-                ess_tail,
-                r_hat,
+        elif stat_focus == "median":
+            diagnostics = xr.apply_ufunc(
+                _make_ufunc(_multichain_statistics, n_output=4, ravel=False),
+                dataset,
+                kwargs=dict(focus="median"),
+                input_core_dims=(("chain", "draw"),),
+                output_core_dims=tuple([] for _ in range(4)),
             )
-        elif kind == "stats":
-            metrics_ = (mean, sd, hdi_lower, hdi_higher)
-            metrics_names_ = metrics_names_[:4]
-        elif kind == "diagnostics":
-            metrics_ = (mcse_mean, mcse_sd, ess_bulk, ess_tail, r_hat)
-            metrics_names_ = metrics_names_[4:]
-        metrics.extend(metrics_)
-        metric_names.extend(metrics_names_)
+            diagnostics_names: Tuple[str, ...] = (
+                "mcse_median",
+                "ess_median",
+                "ess_tail",
+                "r_hat",
+            )
+        metrics.extend(diagnostics)
+        metric_names.extend(diagnostics_names)
 
     if circ_var_names:
         if kind != "diagnostics":
-            for metric, circ_stat in zip(
-                # Replace only the first 5 statistics for their circular equivalent
-                metrics[:5],
-                (circ_mean, circ_sd, circ_hdi_lower, circ_hdi_higher, circ_mcse),
-            ):
+            if stat_focus == "mean":
+                for metric, circ_stat in zip(
+                    # Replace only the first 5 statistics for their circular equivalent
+                    metrics[:5],
+                    (circ_mean, circ_sd, circ_hdi_lower, circ_hdi_higher, circ_mcse),
+                ):
+                    for circ_var in circ_var_names:
+                        metric[circ_var] = circ_stat[circ_var]
+            else:
```

Comment:
I'll try to come back to interaction between stat_focus and circular variables at some point next week.

Reply:
After some deliberation it is best if we (at least for now) raise an error if `stat_focus!="mean"` and `circ_var_names in not None` (which should happen at the beginning, like the check to ensure stat_focus is mean or median.